source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -0,0 +1,117 @@
+#define NOMINMAX
+
+#include <custom/xnetsocket_converters.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+using namespace nixnetsocket_grpc;
+using namespace ::testing;
+namespace pb = ::google::protobuf;
+
+namespace ni {
+namespace tests {
+namespace unit {
+namespace {
+TEST(XnetConvertersTests, IPv4GrpcSockAddr_ConvertFromGrpc_CreatesIPv4NXSockAddr)
+{
+  constexpr auto PORT = 1234;
+  constexpr auto ADDRESS = 0xAABBCCDD;
+  auto grpc_sock_addr = SockAddr{};
+  grpc_sock_addr.mutable_ipv4()->set_port(PORT);
+  grpc_sock_addr.mutable_ipv4()->set_addr(ADDRESS);
+
+  auto converted_addr = convert_from_grpc<nxsockaddr>(grpc_sock_addr);
+  auto addr_ptr = static_cast<nxsockaddr*>(converted_addr);
```

Comment:
Note: `static_cast` tests the implicit conversion. `reinterpret_cast` is what the driver would do given the address family.